src/dfx/src/commands/canister/deposit_cycles.rs

```diff
@@ -64,31 +100,70 @@ pub async fn exec(
 
     let proxy_sender;
 
-    // choose default wallet if no wallet is specified
     if call_sender == &CallSender::SelectedId {
-        let wallet = get_or_create_wallet_canister(
+        match get_or_create_wallet_canister(
             env,
             env.get_network_descriptor(),
             env.get_selected_identity().expect("No selected identity"),
         )
-        .await?;
-        proxy_sender = CallSender::Wallet(*wallet.canister_id_());
-        call_sender = &proxy_sender;
+        .await
+        {
+            Ok(wallet) => {
+                proxy_sender = CallSender::Wallet(*wallet.canister_id_());
+                call_sender = &proxy_sender;
+            }
+            Err(err) => {
+                if CYCLES_LEDGER_ENABLED && matches!(err, crate::lib::identity::wallet::GetOrCreateWalletCanisterError::NoWalletConfigured { .. }) {
```

Comment:
Why does the error handling of the `get_or_create_wallet_canister` depend on the flag `CYCLES_LEDGER_ENABLED`?

Reply:
Because until we enable the cycles ledger functionality we want to preserve the previous behavior of erroring out because no cycles wallet is configured. Once the cycles ledger is enabled we will just try to use the cycles in the cycles ledger